packages/Webkul/Product/src/Type/Bundle.php

```diff
@@ -542,29 +542,32 @@ public function getAdditionalOptions($data)
     {
         $bundleOptionQuantities = $data['bundle_option_qty'] ?? [];
 
-        foreach ($data['bundle_options'] as $optionId => $optionProductIds) {
-            $option = $this->productBundleOptionRepository->find($optionId);
+        $productBundleOptions = $this->productBundleOptionRepository
+            ->whereIn('id', array_keys($data['bundle_options']))
+            ->orderBy('sort_order')
+            ->get();
 
+        foreach ($productBundleOptions as $option) {
             $labels = [];
 
-            foreach ($optionProductIds as $optionProductId) {
+            foreach ($data['bundle_options'][$option->id] as $optionProductId) {
                 if (! $optionProductId) {
                     continue;
                 }
 
                 $optionProduct = $this->productBundleOptionProductRepository->find($optionProductId);
 
-                $qty = $data['bundle_option_qty'][$optionId] ?? $optionProduct->qty;
+                $qty = $data['bundle_option_qty'][$option->id] ?? $optionProduct->qty;
 
-                if (! isset($data['bundle_option_qty'][$optionId])) {
-                    $bundleOptionQuantities[$optionId] = $qty;
+                if (! isset($data['bundle_option_qty'][$option->id])) {
+                    $bundleOptionQuantities[$option->id] = $qty;
                 }
 
                 $labels[] = $qty . ' x ' . $optionProduct->product->name . ' ' . core()->currency($optionProduct->product->getTypeInstance()->getMinimalPrice());
             }
 
             if (count($labels)) {
-                $data['attributes'][$option->id] = [
+                $data['attributes'][] = [
```

Comment:
NOTE: I removed the array being indexed by the bundle option ID, which was necessary to fix the ordering. I could not find the index being used anywhere, but someone with more knowledge should review this.